pkg/router/metrics/haproxy/haproxy.go

```diff
@@ -614,7 +619,7 @@ func exportCSVFields(csvParseFailures prometheus.Counter, metrics metrics, basel
 			}
 			value += baselineValues[fieldIdx]
```

Comment:
Do we need a nil check here as well?  Would it be better to avoid passing in a nil value?  I might be missing subtlety in how these functions all fit together, but focusing on the contract of `exportCSVFields`, I think we want always to pass `e.baseCounterValues[rowID]` if we have it or a new empty map if we don't as well as a separate argument (such as the erstwhile `updateBaseline`) to tell `exportCSVFields` whether to update the provided `baselineValues`.

A shortcut to avoid adding an additional parameter would be to pass an empty map for `counterMetrics` if updates are not wanted.  (But I think we need to pass in `baselineValues` in any case.)

Reply:
baseline for nil is zero.  So no, we don't need a nil check.

Reply:
I mean, in the latest version of the PR, `exportAndRecordRow` initializes `baselineValues`:
```go
	var baselineValues map[int]int64
￼```
and may leave it uninitialized:
```go
	if targetValues != nil {
		baselineValues = e.baseCounterValues[rowID]
		if baselineValues == nil {
			baselineValues = make(map[int]int64)
			targetValues[rowID] = baselineValues
		}
	}
```
before passing `baselineValues` to `exportCSVFields`:
```go
	exportCSVFields(e.csvParseFailures, metrics, baselineValues, e.counterMetrics, csvRow, labels)
```
which indexes `baselineValues` without checking that it is non-nil:
```go
			value += baselineValues[fieldIdx]
```
which would cause a run-time panic if `baselineValues` were nil.

Reply:
No, you can read from a nil map and it returns the default zero value.  Anyway, it switched to an array.